### ISSUE TITLE
Code insights "Update default filters" being disabled while no edits have been made is potentially confusing

### ISSUE THREAD

**Issue by felixfbecker** [sourcegraph/sourcegraph#24081]
![image](https://user-images.githubusercontent.com/10532611/129886183-950c44dc-0eb6-4fca-9504-4acd735fc300.png)

Disabling submit buttons is generally considered bad practice, as it doesn't allow the user to see why it is disabled (a submit button would normally trigger form validation and focus the first invalid/missing field): https://www.smashingmagazine.com/2021/08/frustrating-design-patterns-disabled-buttons/

What is confusing in this form especially is that the "Save as new view" button is _not_ disabled – is there a reason for that?

**Comment by vovakulikov** [comment]
@AlicjaSuska I also think that we don't need to disable that button. Do you think we could remove disable loading? What is the original reason to disable it?

**Comment by AlicjaSuska** [comment]
There are couple things to discuss here:
1. We have two types of buttons that save default filters:
- ['set default filters'](https://www.figma.com/file/1zvF8Owrh1ndVH29lvyrXn/RFC-419:-Code-Insights-“Drilldown”?node-id=1109%3A12000) - if there are no default filters saved yet
- ['update default filters'](https://www.figma.com/file/1zvF8Owrh1ndVH29lvyrXn/RFC-419:-Code-Insights-“Drilldown”?node-id=1266%3A11563) - if there are default filters set already
There is no 'save default filters' button (shown in the screenshot above).

2. The original logic:
- button disabled - no changes to default filters made, nothing to save
- button active - user made changes to default filters, they can default update filters

3. Alternatively, we can:
-  If the disabled state is confusing, we can hide buttons by default and show them only once some changes to the filters have been made.
- keep the disabled state of the button and add a tooltip that says 'no changes to save'

I think that hiding and showing buttons would be better here, it would catch the user's attention and we would avoid the 'tooltip on tooltip' situation. @vovakulikov @felixfbecker @Joelkw wdyt?

**Comment by Joelkw** [comment]
I don't think that it's confusing because it's common practice for a "save" or "update" button to be disabled before input/changes have been made. 

I worry that hiding buttons (as a general UX pattern) makes uses not aware these things exist until you take action. I would prefer that they do not start out as hidden IMO – I also think it'd make things jumpier visually.

**Comment by AlicjaSuska** [comment]
Good point @Joelkw. I've also talked to @quinnkeast, here are 2 main points for leaving the disabled button as-is:
-buttons/functionality should stay present because it boosts confidence. there’s a difference between “the thing I’m looking for is missing” versus “the thing I’m looking for is disabled” and the latter is better for usability
- all that said, when it comes to “should buttons be shown disabled or not,” the most important thing is that the behavior is consistent everywhere

I agree with the above. I still see the value in the simplicity of hiding/showing the button but, given the arguments above, keeping the disabled state may be better for consistency.

That being said, I feel like our buttons don't really look very disabled. This may be an issue that causes doubt in the first place. I've posted about the possible solution (desaturating the color) on [Slack](https://sourcegraph.slack.com/archives/C0HMGV90V/p1633436088119700)

@vovakulikov I've also noticed the misalignment between the design system, my design and the implementation: disabled button background should be Semantic/body-bg not Semantic/body-bg-1 - this may also help make the button look a little less 'enabled' by reducing the contrast. I think we've talked about it in the past, I apologise if designs were misleading.

**Comment by AlicjaSuska** [comment]
@felixfbecker regarding disabling / enabling 'save as new view' button:
- my initial thinking was: user can 'save as new view' and then modify the filters and save the configuration using 'update filters' button
- I see your point here that it may be confusing, I'm open to changing the logic to disabling this button unless user makes changes to filters
@vovakulikov @Joelkw wdyt?

**Comment by unclejustin** [comment]
Decided to go ahead and remove disable logic.
